pandas/core/indexes/multi.py

```diff
@@ -3095,8 +3103,54 @@ def _update_indexer(idxr, indexer=indexer):
         # empty indexer
         if indexer is None:
             return Int64Index([])._ndarray_values
+
+        if need_sort:
+            indexer = self._reorder_indexer(seq, indexer)
+
         return indexer._ndarray_values
 
+    def _reorder_indexer(self, seq, indexer: Int64Index) -> Int64Index :
+        """
+        Reorder an indexer of a MultiIndex (self) so that the label are in the
+        same order as given in seq
+
+        Parameters
+        ----------
+        seq : label/slice/list/mask or a sequence of such
```

Comment:
We iterate over `seq`, so presumably it can't be a label or a slice. Can you confirm that it's just listlike or arraylike?

Reply:
This is the same argument as in .get_locs, it is always a list, sometime with only one element, and the element can be a slice/label/list/mask for one level of the multi index.

Reply:
I think we just use ArrayLike here (can you try to type seq as well)

Reply:
Ok for ArrayLike.
For seq, I logged the type passed when testing indexes and indexing, and it is a tuple that can contains the following type:
```
<class 'int'>
<class 'list'>
<class 'numpy.ndarray'>
<class 'pandas.core.indexes.numeric.Int64Index'>
<class 'pandas.core.series.Series'>
<class 'set'>
<class 'slice'>
<class 'str'>
<class 'tuple'>
```
So the type of seq would be something like 
`Tuple[Scalar, AnyArrayLike, Iterable, …] `
Does that make sense?